### PR TITLE
fix(kafka create): run interactive if no name provided and fix typos

### LINKED ISSUE #703: `rhoas kafka create` should prompt for name if not provided
## Description

`rhoas kafka create` doesn't run in interactive mode as expected.

**Version**: 0.24.3

### Steps to reproduce

Run `rhoas kafka create --provider aws`. It should output:
> Error: name does not match ^[a-z]([-a-z0-9]*[a-z0-9])?$

### Expected vs actual behaviour

The cli should prompt for name but instead sends request with name as `""`

### Logs

Screenshot from 2021-06-04 17-02-15

### PR DESCRIPTION
**Pull request by rkpattnaik780**
- Open interactive mode when `rhoas kafka create` is run with flags but no name.
- Fix faulty error message while filtering consumergroups by topic.

Closes #703, #704

### Verification Steps

1. Run `rhoas kafka create --provider aws`, it should open the interactive mode.
2. Run `rhoas kafka consumergroup list --topic <topic-name>` where the topic has no associated consumergroup, it should display the message `Kafka instance "<instance name>" has no consumer groups for topic "<topic-name>"`

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by craicoverflow** [comment]
image

It looks like the **Closing** keyword does not link multiple issues automatically. In this case you can manually link them in the side bar.

**Review by craicoverflow — COMMENTED** [review]
When I pass the `--provider` flag, it will re-prompt me to enter the provider value in the interactive prompt:

```shell
❯ ./rhoas kafka create --provider aws
? Name: my-kafka
? Cloud Provider:  [Use arrows to move, type to filter]
> aws
```

The way it was (supposed to be) working before now was as follows: 

If no name argument is passed, and any value flag _is_ passed, such as provider, then throw an error as the user is trying to mix interactive with non-interactive Kafka creation.

I think that this is more maintainable in the long run, and I believe it is consistent with the other **create** commands (not sure though).

**Comment by rkpattnaik780** [comment]
> I think that this is more maintainable in the long run, and I believe it is consistent with the other **create** commands (not sure though).

Had used `topic create` as a reference. We should update that one later as well.
Screenshot from 2021-06-09 16-06-18

File: pkg/localize/locales/en/cmd/kafka_create.en.toml [hunk 1]
```diff
@@ -69,6 +69,9 @@ one = "Geographical region where the Kafka instance will be deployed"
 [kafka.create.argument.name.error.requiredWhenNonInteractive]
 one = 'name is required. Run "rhoas kafka create <name>"'
 
+[kafka.create.flags.notAllowedWhenInteractive]
+one = 'region and provider flags not allowed for interactive Kafka instance creation'
```
Comment (rkpattnaik780) [inline pkg/localize/locales/en/cmd/kafka_create.en.toml:73]: Suggestion for a better error message @craicoverflow ?

Comment (craicoverflow) [reply pkg/localize/locales/en/cmd/kafka_create.en.toml:73]: Swap it round like something like this: `"name" argument is required when creating a Kafka instance`

Comment (rkpattnaik780) [reply pkg/localize/locales/en/cmd/kafka_create.en.toml]: Changed it to `'name is required. Run "rhoas kafka create <name>"'`

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**